copy/copy.go

```diff
@@ -121,6 +144,23 @@ type Options struct {
 	Progress         chan types.ProgressProperties // Reported to when ProgressInterval has arrived for a single artifact+offset.
 	// manifest MIME type of image set by user. "" is default and means use the autodetection to the the manifest MIME type
 	ForceManifestMIMEType string
+	// CheckAuthorization is an indication if the image does not need to be decrypted, but only
+	// requires checking if the it can be decrypted.
+	// This is an optimization for cached images to see if provided keys have the authorization
+	// to use the cached content without having to decrypt the layers again.
+	CheckAuthorization bool
```

Comment:
>The Skopeo PR containers/skopeo#732 does not use the field. Is that something we really need? How is it intended to be used?

This is intended to be used by cri-o. In kubernetes, the creation of container should be given the option to check if the image is decryptable if it already exists in the cache. Just to make sure the user is able to use the image.

Reply:
Thanks! We really need a test for this in the Skopeo PR to prevent us from silently regressing in the future. But before doing that, let's find consensus on the API.

Reply:
Linking the relevant KEP to the thread: https://github.com/kubernetes/enhancements/pull/1066

This picture is helpful:

![image](https://user-images.githubusercontent.com/3060102/66320353-b0f0ae80-e8ec-11e9-96f4-fdf90db1edd8.png)